code/game/vehicles/spacepods/spacepod.dm

```diff
@@ -464,129 +468,119 @@
+		return 0
 
 	for(var/mob/living/carbon/slime/S in range(1,usr))
 		if(S.Victim == user)
 			user << "You're too busy getting your life sucked out of you."
-			return
-
-	try_user_enter(user)
-
-/obj/spacepod/proc/try_user_enter(var/mob/user)
-	if(!src.occupant && !src.occupant2)
-		visible_message("<span class='notice'>[user] starts to climb into [src.name]</span>")
-		if(enter_after(40,user))
-			if(!src.occupant)
-				moved_inside(user)
-			else if(src.occupant != user)
-				user << "<span class='notice'>[src.occupant] was faster. Try better next time, loser.</span>"
-		else	user << "<span class='notice'>You stop entering the spacepod.</span>"
-		return
-
-	if(!allow2enter)
-		user << "<span class='danger'>The door is locked!</span>"
-		return
+			return 0
 
-	if(occupant && occupant2)
-		user << "<span class='danger'>You can't fit in \the [src], it's full!</span>"
-		return
+	move_inside(user)
 
-	if(occupant && !occupant2)
-		user << "<span class='notice'><b>You start climbing into the passenger bay.</b></span>"
-		user_enter(user)
+/obj/spacepod/proc/move_inside(mob/user)
+	if(!istype(user))
+		log_debug("SHIT'S GONE WRONG WITH THE SPACEPOD [src] AT [x], [y], [z], AREA [get_area(src)], TURF [get_turf(src)]")
 
-	if(!occupant && occupant2)
-		user << "<span class='notice'><b>You start climbing into the pilot's seat.</b></span>"
-		user_enter(user)
+	__SANITY_CHECK_FUCKING_OCCUPANT_VARS()
 
-/obj/spacepod/proc/user_enter(var/mob/user)
-	visible_message("<span class='notice'>[user] starts to climb into \the [src.name].</span>")
-	if(enter_after(40,user))
-		moved_inside(user)
+	if(!occupant)
+		visible_message("<span class='notice'>[user] starts to climb into \the [src].</span>")
+		if(do_after(user, 40, target = src))
+			if(!occupant)
+				user.stop_pulling()
+				occupant = user
+				user.forceMove(src)
+				add_fingerprint(user)
+				playsound(src, 'sound/machines/windowdoor.ogg', 50, 1)
+			else
+				user << "<span class='notice'>[occupant] was faster. Try better next time, loser.</span>"
+		else
+			user << "<span class='notice'>You stop entering \the [src].</span>"
+
+	else if(!occupant2)
+		visible_message("<span class='notice'>[user] starts to climb into \the [src].</span>")
+		if(do_after(user, 40, target = src))
+			if(!occupant2)
+				user.stop_pulling()
+				occupant2 = user
+				user.forceMove(src)
+				add_fingerprint(user)
+				playsound(src, 'sound/machines/windowdoor.ogg', 50, 1)
+			else
+				user << "<span class='notice'>[occupant] was faster. Try better next time, loser.</span>"
+		else
+			user << "<span class='notice'>You stop entering \the [src].</span>"
 	else
-		user << "You stop entering the spacepod."
+		user << "<span class='danger'>You can't fit in \the [src], it's full!</span>"
+
+/obj/spacepod/proc
+	__SANITY_CHECK_FUCKING_OCCUPANT_VARS()	// /obj/spacepod/proc/__SANITY_CHECK_FUCKING_OCCUPANT_VARS()
+		if(occupant)
+			if(!ismob(occupant))
+				occupant = null
+				occupant.forceMove(get_turf(src))
```

Comment:
shouldn't occupant be nulled second? What this effectively null out the reference then end up not moving someone to the turf?